sdks/python/src/opik/evaluation/models/litellm/litellm_chat_model.py

```diff
@@ -193,12 +279,33 @@ def generate_provider_response(
         ):
             all_kwargs = opik_monitor.try_add_opik_monitoring_to_params(all_kwargs)
 
-        response = self._engine.completion(
-            model=self.model_name, messages=messages, **all_kwargs
+        retries = kwargs.pop("__opik_retries", 3)
+        try:
+            attempts = max(1, int(retries))
+        except (TypeError, ValueError):
+            attempts = 1
+        backoff = 0.5
+
+        last_exception: Optional[Exception] = None
+        for attempt in range(attempts):
+            try:
+                return self._engine.completion(
+                    model=self.model_name, messages=messages, **all_kwargs
+                )
+            except Exception as exc:  # noqa: BLE001
+                last_exception = exc
+                if attempt == attempts - 1:
+                    raise
+                time.sleep(backoff)
+                backoff *= 2
+
+        if last_exception is not None:
+            raise last_exception
+        raise RuntimeError(
+            "LLM completion failed without executing any attempts; "
```

Comment:
We’ve recently introduced `exceptions.BaseLLMError` to the SDK. It might be a better fit to raise here instead of using the generic `RuntimeError`.